dot/parachain/candidate-validation/candidate_validation_test.go

```diff
@@ -612,22 +617,15 @@ func Test_precheckPvF(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			t.Parallel()
 
-			toSubsystem := make(chan any)
 			candidateValidationSubsystem := CandidateValidation{
 				pvfHost:    newValidationHost(),
 				BlockState: mockBlockState,
 			}
-			defer candidateValidationSubsystem.Stop()
-
-			go candidateValidationSubsystem.Run(context.Background(), toSubsystem)
-
-			sender := make(chan PreCheckOutcome)
-			tt.msg.ResponseSender = sender
-
-			toSubsystem <- tt.msg
-			result := <-sender
-
+			result, err := candidateValidationSubsystem.precheckPvF(tt.msg.RelayParent, tt.msg.ValidationCodeHash)
 			require.Equal(t, tt.expectedResult, result)
+			if tt.expectedError != nil {
+				require.EqualError(t, err, tt.expectedError.Error())
+			}
```

Comment:
Also, add an else block to ensure the `err` is nil when `tt.expectedError` is nil.

Reply:
Added.